### PR TITLE
Integration tests infrastructure - integration tests part 1

### PR DESCRIPTION
**Pull request by iulianpascalau**
- added integration tests infrastructure

### CODE REVIEW

File: integrationTests/mock/elrondChainMock.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+		LatestTagSoftwareVersion: "",
+		MinGasPrice:              1000000000,
+		MinTransactionVersion:    1,
+	}, nil
+}
+
+// SendTransaction -
+func (mock *ElrondChainMock) SendTransaction(transaction *data.Transaction) (string, error) {
+	if transaction == nil {
+		panic("nil transaction")
```
Comment (dragos-rebegea) [inline integrationTests/mock/elrondChainMock.go:46]: why panic and not returning error ?

**Review by dragos-rebegea — APPROVED** [review]

File: integrationTests/mock/elrondChainMock.go (hunk 1 above)
Comment (sasurobert) [reply integrationTests/mock/elrondChainMock.go:46]: +1

File: integrationTests/mock/elrondChainMock.go [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+	}
+
+	addrAsBech32 := transaction.SndAddr
+	addressHandler, err := data.NewAddressFromBech32String(addrAsBech32)
+	if err != nil {
+		panic(fmt.Sprintf("%v while creating address handler for string %s", err, addrAsBech32))
```
Comment (sasurobert) [inline integrationTests/mock/elrondChainMock.go:52]: Panic again?

**Review by sasurobert — COMMENTED** [review]
No panics?
How hard would it be to actually integrate execution from elrond-go? Like txProcessor and context from integration tests?

**Review by sasurobert — APPROVED** [review]

**Review by dragos-rebegea — APPROVED** [review]

**Pull request merged**